### PR TITLE
Make user model parcelable

### PR DESCRIPTION
**Pull request by ezaquarii**
Most of the implementation has been salvaged from https://github.com/nextcloud/android/pull/5006

### CODE REVIEW

File: src/androidTest/java/com/nextcloud/client/account/RegisteredUserTest.kt [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+                server = server
+            )
+        }
+    }
+
+    private lateinit var user: RegisteredUser
+
+    @Before
+    fun setUp() {
+        user = buildTestUser("test.nextcloud.localhost.localdomain")
```
Comment (tobiasKaminsky) [inline src/androidTest/java/com/nextcloud/client/account/RegisteredUserTest.kt]: That is not a valid accountName, it should be test@nextloud.localhost.localdomain.

File: src/androidTest/java/com/nextcloud/client/account/AnonymousUserTest.kt [hunk 2]
```diff
@@ -0,0 +1,33 @@
+package com.nextcloud.client.account
```
Comment (tobiasKaminsky) [inline src/androidTest/java/com/nextcloud/client/account/AnonymousUserTest.kt]: License

File: src/androidTest/java/com/nextcloud/client/account/RegisteredUserTest.kt [hunk 3]
```diff
@@ -0,0 +1,99 @@
+package com.nextcloud.client.account
```
Comment (tobiasKaminsky) [inline src/androidTest/java/com/nextcloud/client/account/RegisteredUserTest.kt]: License

**Comment by tobiasKaminsky** [comment]
```
Process: com.nextcloud.client, PID: 19057
    java.lang.SecurityException: uid 10304 cannot get user data for accounts of type: com.google
        at android.os.Parcel.createException(Parcel.java:2071)
        at android.os.Parcel.readException(Parcel.java:2039)
        at android.os.Parcel.readException(Parcel.java:1987)
   at android.accounts.AccountManager.getUserData(AccountManager.java:514)
        at com.nextcloud.client.etm.EtmViewModel.getAccounts(EtmViewModel.kt:80)
        at com.nextcloud.client.etm.pages.EtmAccountsFragment.onResume(EtmAccountsFragment.kt:49)
 ```

I guess that you ask for all accounts, but not only for our accounts.

File: src/test/java/com/nextcloud/client/etm/TestEtmViewModel.kt [hunk 4]
```diff
@@ -54,13 +55,15 @@ class TestEtmViewModel {
...
+        protected lateinit var platformAccountManager: AccountManager
         protected lateinit var sharedPreferences: SharedPreferences
         protected lateinit var vm: EtmViewModel
 
         @Before
         fun setUpBase() {
             sharedPreferences = mock()
-            vm = EtmViewModel(sharedPreferences)
+            platformAccountManager = mock()
+            vm = EtmViewModel(sharedPreferences, platformAccountManager)
```
Comment (tobiasKaminsky) [inline src/test/java/com/nextcloud/client/etm/TestEtmViewModel.kt]: Can we there then have also a screenshot test?
If you do not have time for this, just ping me, and I'll do it.

**Comment by ezaquarii** [comment]
> I guess that you ask for all accounts, but not only for our accounts.

Indeed, this was a bug. I changed that to query only for nextcloud accounts. But honestly, this should probably be handled by `UserAccountManager`.

File: src/androidTest/java/com/nextcloud/client/account/RegisteredUserTest.kt (hunk 1 above)
Comment (ezaquarii) [reply src/androidTest/java/com/nextcloud/client/account/RegisteredUserTest.kt]: account name is fixed

File: src/androidTest/java/com/nextcloud/client/account/AnonymousUserTest.kt (hunk 2 above)
Comment (ezaquarii) [reply src/androidTest/java/com/nextcloud/client/account/AnonymousUserTest.kt]: added licence

File: src/androidTest/java/com/nextcloud/client/account/RegisteredUserTest.kt (hunk 3 above)
Comment (ezaquarii) [reply src/androidTest/java/com/nextcloud/client/account/RegisteredUserTest.kt]: added licence

File: src/test/java/com/nextcloud/client/etm/TestEtmViewModel.kt (hunk 4 above)
Comment (ezaquarii) [reply src/test/java/com/nextcloud/client/etm/TestEtmViewModel.kt]: @tobiasKaminsky View model does not contain any UI code - which makes unit testing very easy.

If we'd like to have a UI test for this part of the app, we'd need to test `EtmAccountsFragment` (and hosting activity).

Comment (tobiasKaminsky) [reply src/test/java/com/nextcloud/client/etm/TestEtmViewModel.kt]: Ah, I misread.

**Pull request merged**